fastlane/lib/fastlane/new_action.rb

```diff
@@ -1,8 +1,8 @@
 module Fastlane
   # Guides the new user through creating a new action
   module NewAction
-    def self.run
-      name = fetch_name
+    def self.run(new_action_name: nil)
+      name = check_action_name_from_args(new_action_name) ? new_action_name : fetch_name
```

Comment:
Should this be switched to read: 
```
name = new_action_name.nil? ? fetch_name : check_action_name_from_args(new_action_name)
```
The current implementation would print out `Name invalid, try again!` even if the user didn't pass in a name.